### ISSUE TITLE
Pushwoosh Android SDK uses an old firebase version which contains a deprecated class

### ISSUE THREAD

**Issue by elliemehrganrad** [Pushwoosh/pushwoosh-android-sdk#166]
We us Pushwoosh Android SDK `6.2.11` in our app. If I update `com.google.firebase:firebase-messaging` SDK to the latest version which is `22.0.0` the app crashes. The root cause is that Pushwoosh SDK is using `com.google.firebase.iid.FirebaseInstanceId` which is deprecated in the latest version.
Here is the stack trace:

```2021-07-15 08:52:21.133 28757-28757/nl.albumprinter.smartphone.editor.alpha E/Pushwoosh: You must add "implementation 'com.google.firebase:firebase-messaging:+'" line to your app build.gradle.
    Check documentation for more info https://goo.gl/UVJKfp
2021-07-15 08:52:21.135 28757-28757/nl.albumprinter.smartphone.editor.alpha E/AndroidRuntime: FATAL EXCEPTION: main
    Process: nl.albumprinter.smartphone.editor.alpha, PID: 28757
    java.lang.RuntimeException: Unable to get provider com.pushwoosh.PushwooshInitProvider: java.lang.IllegalStateException: You must add "implementation 'com.google.firebase:firebase-messaging:+'" line to your app build.gradle.
    Check documentation for more info https://goo.gl/UVJKfp
        at android.app.ActivityThread.installProvider(ActivityThread.java:7724)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:7260)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7156)
        at android.app.ActivityThread.access$2200(ActivityThread.java:292)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2195)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:213)
        at android.app.ActivityThread.main(ActivityThread.java:8147)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1101)
     Caused by: java.lang.IllegalStateException: You must add "implementation 'com.google.firebase:firebase-messaging:+'" line to your app build.gradle.
    Check documentation for more info https://goo.gl/UVJKfp
        at com.pushwoosh.firebase.internal.checker.FirebaseChecker.check(Unknown Source:15)
        at com.pushwoosh.firebase.internal.registrar.FcmRegistrar.init(Unknown Source:5)
        at com.pushwoosh.notification.PushwooshNotificationManager.d(Unknown Source:2)
        at com.pushwoosh.d.a(Unknown Source:59)
        at com.pushwoosh.PushwooshPlatform.l(Unknown Source:2)
        at com.pushwoosh.PushwooshInitializer.init(Unknown Source:67)
        at com.pushwoosh.PushwooshInitProvider.onCreate(Unknown Source:18)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:2121)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:2094)
        at android.app.ActivityThread.installProvider(ActivityThread.java:7719)
        	... 10 more
     Caused by: java.lang.ClassNotFoundException: com.google.firebase.iid.FirebaseInstanceId
        at java.lang.Class.classForName(Native Method)
        at java.lang.Class.forName(Class.java:454)
        at java.lang.Class.forName(Class.java:379)
        at com.pushwoosh.firebase.internal.checker.FirebaseChecker.check(Unknown Source:2)
        	... 19 more
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.iid.FirebaseInstanceId" on path: DexPathList[[zip file "/data/app/nl.albumprinter.smartphone.editor.alpha-AiDDm5t52mqcXUGNf4fCsw==/base.apk"],nativeLibraryDirectories=[/data/app/nl.albumprinter.smartphone.editor.alpha-AiDDm5t52mqcXUGNf4fCsw==/lib/arm64, /data/app/nl.albumprinter.smartphone.editor.alpha-AiDDm5t52mqcXUGNf4fCsw==/base.apk!/lib/arm64-v8a, /system/lib64, /system/product/lib64, /hw_product/lib64, /system/product/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:196)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        	... 23 more

**Comment by wfhm** [comment]
Duplicate of #163. This will be fixed in upcoming releases, meanwhile you can add the missing dependency manually:

```implementation "com.google.firebase:firebase-iid:21.1.0"```

**Comment by elliemehrganrad** [comment]
Thanks @wfhm

**Comment by wfhm** [comment]
@elliemehrganrad Fixed in 6.3.0.

**Comment by elliemehrganrad** [comment]
@wfhm Thank you for letting me know
